### PR TITLE
Move common CSI driver RPCs from package connection -> rpc

### PR DESCRIPTION
**Pull request by andrewsykim**
Before updating a bunch of CSI sidecars to use the common CSI RPCs `GetDriverName`, `GetPluginCapabilities`, `GetControllerCapabilities` and `Probe`, I wanted to propose that we move these to its own package called `rpc`. This keeps the `connection` package cleaner and only hold code related to connection management. 

/assign @msau42 @jsafrane

### CODE REVIEW

**Comment by jsafrane** [comment]
Do we care about backward compatibility? It will break anyone who uses this library. 

Otherwise LGTM.

File: rpc/common_test.go [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+	probeCalls     []probeCall
+	probeCallCount int
+}
+
+type probeCall struct {
+	response *csi.ProbeResponse
+	err      error
+}
+
+var _ csi.IdentityServer = &identityServer{}
```
Comment (msau42) [inline rpc/common_test.go]: nit: can you prepend "fake" to the name

File: rpc/common_test.go [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+	i.probeCallCount++
+	return call.response, call.err
+}
+
+type controllerServer struct {
+	controllerGetCapabilitiesResponse *csi.ControllerGetCapabilitiesResponse
+	err                               error
+}
+
+var _ csi.ControllerServer = &controllerServer{}
```
Comment (msau42) [inline rpc/common_test.go]: nit: can you prepend "fake" to the name

File: rpc/common_test.go (hunk 1 above)
Comment (andrewsykim) [reply rpc/common_test.go]: done!

File: rpc/common_test.go (hunk 2 above)
Comment (andrewsykim) [reply rpc/common_test.go]: done!

**Comment by andrewsykim** [comment]
Thanks @msau42! Can you let me know when a new release is cut with this change?

**Comment by msau42** [comment]
https://github.com/kubernetes-csi/csi-lib-utils/releases/tag/v0.4.0-rc1

**Pull request merged**